docs/develop/01_sdk/02_cookbook/05_messaging/01_messaging.md

```diff
@@ -0,0 +1,57 @@
+---
+id: messaging_book
+title: Generate a Message
+---
+
+import TsJsBlock from '@site/src/components/TsJsBlock';
+
+import GenerateMessage from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/messaging/01_generate_message.ts';
+import EncryptMessage from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/messaging/02_encrypt_message.ts';
+import DecryptMessage from '!!raw-loader!@site/code_examples/sdk_examples/src/core_features/messaging/03_decrypt_message.ts';
+
+KILT defines a [unicast](https://en.wikipedia.org/wiki/Unicast) [messaging protocol](../../../../concepts/06_messaging.md).
+Each of the messages sent are encrypted using the [DID key agreement key](https://www.w3.org/TR/did-core/#key-agreement).
+The key agreement key of a DID can be queried from the KILT blockchain.
+
+A message consists of the senders DID URI, the receivers DID URI, the message type and the body.
+There are multiple different message types.
+In this example we are going to build a `request-credential` message.
+The message structure is checked and validated on by the KILT SDK to ensure the users are sending correctly structured messages.
+
+The following example here will generate a message by constructing the message content.
+The message content includes a valid `cTypeHash` and a list of `trusted attesters`.
+The message requires a `messageBody`, sender and receiver uri.
```

Comment:
Same as above, we talk about keys and not DID URIs here.

Reply:
When we send a message, we include the URI of the sender and receiver.